MyMovies/Model Controllers/MovieController.swift

```diff
@@ -57,8 +66,118 @@ class MovieController {
+    guard let uuid = movie.identifier else {
+    completion(.failure(.noIdentifier))
+    return
+    }
+    let requestURL = firebaseURL.appendingPathComponent(uuid.uuidString).appendingPathExtension("json")
+    var request = URLRequest(url: requestURL)
+    request.httpMethod = "DELETE"
+    
+    let task = URLSession.shared.dataTask(with: request) { (data, _, error) in
+    completion(.success(true))
+    }
+    task.resume()
+    }
+    
+    private func updateMovies(with representations: [MovieRepresentation]) throws {
+        let context = CoreDataStack.shared.container.newBackgroundContext()
+        let identifiersToFetch = representations.compactMap({UUID(uuidString: $0.identifier)})
+        let representationsByID = Dictionary(uniqueKeysWithValues: zip(identifiersToFetch, representations))
+        var moviesToCreate = representationsByID
+        let fetchRequest: NSFetchRequest<Movie> = Movie.fetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "identifier IN %@", identifiersToFetch)
+        context.performAndWait {
+            do {
+                let existingMovies = try context.fetch(fetchRequest)
+                for movie in existingMovies {
+                    guard let id = movie.identifier,
+                        let representation = representationsByID[id] else {
+                            continue }
+                    movie.title = representation.title
+                    movie.hasWatched = representation.hasWatched
+                    moviesToCreate.removeValue(forKey: id)
+                }
+                for representation in moviesToCreate.values {
+                    Movie(movieRepresentation: representation, context: context)
+                }
+            } catch {
+                print("Error fetching movies for UUIDs: \(error)")
+            }
+        }
+        try CoreDataStack.shared.save(context: context)
+    }
+    
+    func sendMovieToServer(movie: Movie, completion: @escaping CompletionHandler = { _ in }) {
+    guard let uuid = movie.identifier else {
+    completion(.failure(.noIdentifier))
+    return
+    }
+    let requestURL = firebaseURL.appendingPathComponent(uuid.uuidString).appendingPathExtension("json")
+    var request = URLRequest(url: requestURL)
+    request.httpMethod = "PUT"
+    
+    do {
+    guard let representation = movie.movieRepresentation else {
+    completion(.failure(.noRep))
+    return
+    }
+    request.httpBody = try JSONEncoder().encode(representation)
+    } catch {
+    print("Error encoding movie: \(error)")
+    completion(.failure(.noEncode))
+    return
+    }
+    
+    let task = URLSession.shared.dataTask(with: request) { (_, _,  error) in
+    if let error = error {
+    print ("Error PUTting movie to server: \(error)")
+    completion(.failure(.otherError))
+    return
+    }
+    completion(.success(true))
+    }
+    task.resume()
```

Comment:
control-i please 😆 I know you normally have everything indented nicely, not going to hold this against you.